### ISSUE TITLE
Execution problem

### ISSUE THREAD

**Issue by uwadeliu** [bimobject/public-api-desktop-app-proof-of-concept#1]
Hi,
I'm trying to build and execute the application. But have some issues.
First, I can't compile without changes.
What kind of development environment and version have you tried this on?

On https://developer.bimobject.com I have created an app which use
authorizationCodeWithProofKey
redirect url: http://127.0.0.1/
When I manage to compile, and run the application, I get an exeption in file
ApiRequestHelper.cs at row 35, see below snippet.

Is it possible for you to guide me, to get this example code working?
What kind of error it is, and what to do about it?
Best regards
uwadeliu

try
{
    // gets the response
    var response = await request.GetResponseAsync();

Execption:
Message	"Fjärrservern returnerade ett fel: (400) Felaktig begäran."
Status ProtocolError

**Comment by c2TheOlin** [comment]
Hi 

Regarding your first issue:
Could you tell me a little more about the changes you had to make? 
Ive just tried it out with VS2015 and see a problem there. Its looks like it could be the C#7 language features causing issues: 
https://stackoverflow.com/questions/39461407/how-to-use-c7-with-visual-studio-2015

I can push a partial fix for this but if using VS2015 you will still see some syntax highlighting.

For the second issue, it seems the request was made with an incorrect protocol. At which point do you get this error? I notice that one of the Url endpoints should be https. Ill correct that at the same time. If you pull the latest changes and give it a try.

**Comment by uwadeliu** [comment]
Hi,
Thanks for quick response.
This could solve my problems, at least this seems to be possible solutions.
I'll have to check what VS I'm using, and what minor changes I made to
compile.
But the exception I got when running the application stated wrong protocol.

I'll get back to you with result, when I have tested the latest commits.



2017-11-29 12:48 GMT+01:00 Wadelius, Urban <urban.wadelius@flir.se>:
